frontend/packages/dev-console/src/components/import/icon/IconSelect.tsx

```diff
@@ -0,0 +1,64 @@
+import * as React from 'react';
+import { Select, SelectOption, SelectVariant } from '@patternfly/react-core';
+import { getIcons } from '@console/internal/components/catalog/catalog-item-icon';
+
+export type IconSelectProps = {
+  value: string;
+  onValueChanged: (value: string) => void;
+  onClear: () => void;
+};
+
+const IconSelect: React.FC<IconSelectProps> = ({ value, onValueChanged, onClear }) => {
+  const [isOpen, setIsOpen] = React.useState(false);
+
+  // Automatically focus and select the input element when the Select was opened.
+  // TODO: This workaround is required because the PatternFly Select variant typeahead do not show the placeholder with image.
+  const selectRef = React.useRef<any>();
+  React.useEffect(() => {
+    const selectElement: HTMLDivElement = selectRef.current?.parentRef.current;
+    const inputElement: HTMLInputElement = selectElement?.querySelector('input');
+    if (isOpen && inputElement) {
+      inputElement.select();
+    }
+  }, [isOpen]);
+
+  const selectOptions = React.useMemo<React.ReactElement[]>(() => {
+    const options: React.ReactElement[] = [];
+
+    getIcons().forEach(({ label, url }) => {
+      options.push(
+        <SelectOption key={label} value={label}>
+          <img src={url} width="24" height="24" alt={label} className="icon" />
+          {label}
+        </SelectOption>,
+      );
+    });
+
+    return options;
+  }, []);
+
+  const onToggle = () => setIsOpen(!isOpen);
+  const onSelect = (_, selection: string) => {
+    onValueChanged(selection);
+    setIsOpen(false);
+  };
+
+  return (
+    <Select
+      ref={selectRef}
+      variant={isOpen ? SelectVariant.typeahead : SelectVariant.single}
+      placeholderText="Select Icon or enter an URL"
+      typeAheadAriaLabel="Select Icon or enter an URL"
```

Comment:
Please make sure any text is matching what UX wants... this is one of those annoying little things that can slip through.

Reply:
Done